### DISCUSSION TITLE
node_env - csrfField is not a function

### DISCUSSION THREAD

**Discussion by TheJltres** [adonisjs/core#2871] [Help] [answered]
Hi!

I have a simple login form with `{{ csrfField() }}`

I get this error:
`{"level":50,"time":1625068242841,"pid":144098,"hostname":"laptop","name":"someTest","msg":"csrfField is not a function"}`
And it shows default 500 error page, server-error.edge

The problem is that I only get this error when on my `.env`file, `NODE_ENV` is set to `testing` otherwise no error is shown and I can submit the form correctly.

PS -
Is this correct?
https://docs.adonisjs.com/guides/application#node-environment
`Exception: E_INVALID_ENV_VALUE: Value for environment variable "NODE_ENV" must be one of "development,production,testing", instead received "staging"`

Thanks for this awesome framework!

**Comment by WeyIin** [comment, marked as answer]
You can disable it in `shield.ts`, search for csrf `enabled: Env.get('NODE_ENV') !== 'testing',` and change to `enabled: true`, and is always gonna be   enabled. Currently disabled in testing.
